packages/shared/src/bounties.ts

```diff
@@ -0,0 +1,2612 @@
+ * - Bounty remains in storage
+ */
+export async function bountyClosureApprovedTest<
+  TCustom extends Record<string, unknown> | undefined,
+  TInitStorages extends Record<string, Record<string, any>> | undefined,
+>(chain: Chain<TCustom, TInitStorages>) {
+  const [client] = await setupNetworks(chain)
+
+  await setupTestAccounts(client, ['alice'])
+
+  const existentialDeposit = client.api.consts.balances.existentialDeposit
+  const bountyValue = existentialDeposit.toBigInt() * BOUNTY_MULTIPLIER
+  const description = 'Test bounty for closure in approved state'
+
+  // Propose a bounty
+  const proposeBountyTx = client.api.tx.bounties.proposeBounty(bountyValue, description)
+  await sendTransaction(proposeBountyTx.signAsync(testAccounts.alice))
+
+  await client.dev.newBlock()
+  const bountyIndex = await getBountyIndexFromEvent(client)
+
+  // Approve the bounty
+  const approveBountyTx = client.api.tx.bounties.approveBounty(bountyIndex)
+  await scheduleInlineCallWithOrigin(client, approveBountyTx.method.toHex(), {
+    Origins: 'Treasurer',
+  })
+
+  await client.dev.newBlock()
+
+  // Verify bounty is in Approved state
+  const approvedBounty = await getBounty(client, bountyIndex)
+  expect(approvedBounty.status.isApproved).toBe(true)
+
+  // Try to close the bounty - should fail
+  const closeBountyTx = client.api.tx.bounties.closeBounty(bountyIndex)
+  await scheduleInlineCallWithOrigin(client, closeBountyTx.method.toHex(), {
+    Origins: 'Treasurer',
+  })
+
+  await client.dev.newBlock()
+
+  await checkSystemEvents(client, { section: 'scheduler', method: 'Dispatched' })
+    .redact({ redactKeys: /task/ })
+    .toMatchSnapshot('scheduler events when closing bounty with approved state fails')
+
+  // check the result of dispatched event
+  const events = await client.api.query.system.events()
+
+  // Find the Dispatched event from scheduler
+  const dispatchedEvent = events.find((record) => {
+    const { event } = record
+    return event.section === 'scheduler' && event.method === 'Dispatched'
+  })
+
+  assert(dispatchedEvent)
+  assert(client.api.events.scheduler.Dispatched.is(dispatchedEvent.event))
+
+  const dispatchedData = dispatchedEvent.event.data
+  expect(dispatchedData.result.isErr).toBe(true)
+
+  // Decode the module error to get human-readable details
+  const dispatchError = dispatchedData.result.asErr
+  assert(dispatchError.isModule)
+  expect(client.api.errors.bounties.UnexpectedStatus.is(dispatchError.asModule)).toBeTruthy()
+
+  // Verify bounty is still in storage and still Approved
+  const bountyAfterFailedClosure = await getBounty(client, bountyIndex)
+  expect(bountyAfterFailedClosure).toBeTruthy()
+  expect(bountyAfterFailedClosure.status.isApproved).toBe(true)
+
+  await client.teardown()
+}
+
+/**
+ * Test: Bounty closure in PendingPayout state (should fail)
+ *
+ * Verifies:
+ * - Bounty closure fails with PendingPayout error when in PendingPayout state (GeneralAdmin must unassign curator first)
```

Comment:
```suggestion
 * - Bounty closure fails with `PendingPayout` error when in `PendingPayout` state (`GeneralAdmin` must unassign curator first)
```

This is pedantry on my part, but backticks (``) when referring to code in comments are preferable, imo.

Reply:
You are right, i am updating it.